marine.c

```diff
@@ -579,6 +579,35 @@ WS_DLL_PUBLIC marine_result *marine_dissect_packet(int filter_id, unsigned char
     return result;
 }
 
+
+WS_DLL_PUBLIC int compile_bpf(char *bpf, struct bpf_program *fcode, char *err_msg) {
+    pcap_t *pc;
+
+    pc = pcap_open_dead(DLT_EN10MB, MIN_PACKET_SIZE);
+    if (pc != NULL) {
+        struct bpf_program temp_fcode;
+        struct bpf_program *used_fcode;
+        int ret_value;
+
+        used_fcode = (fcode != NULL) ? fcode : &temp_fcode;
+        ret_value = pcap_compile(pc, used_fcode, bpf, 0, 0);
+
+        if (ret_value == -1) {
+             strcpy(err_msg, "Failed compiling the BPF");
+        }
+
+        if (fcode == NULL && ret_value == 0) {
```

Comment:
Why are you checking `ret_value` again?
Wouldn't it make more sense to have:
```c
if (ret_value != 0) {
    // bla
} else if (fcode == NULL) {
   // bla
}
```
?